tests/integration/environment_test.py

```diff
@@ -0,0 +1,33 @@
+from __future__ import absolute_import
+from __future__ import unicode_literals
+
+from .. import mock
+from compose.cli.main import TopLevelCommand
+from compose.project import Project
+from tests.integration.testcases import DockerClientTestCase
+
+
+class EnvironmentTest(DockerClientTestCase):
+
+    def test_no_warning_on_missing_host_environment_var_on_exec(self):
+        db = self.create_service('db')
+        project = Project('composetest', [db], self.client)
+        project.up()
+        options = {'ARGS': ['db', 'bash'],
+                   'COMMAND': 'exec'}
+        command = TopLevelCommand(project, options)
+        with mock.patch('compose.config.environment.log') as fake_log:
+            command.exec_command({'--detach': True,
+                                  '--env': [],
+                                  '--index': '1',
+                                  '--privileged': False,
+                                  '--user': None,
+                                  '--workdir': None,
+                                  '--publish': None,
+                                  '--service-ports': None,
+                                  '-T': False,
+                                  'ARGS': [],
+                                  'COMMAND': 'bash',
+                                  'SERVICE': 'db'})
+            assert fake_log.warn.call_count == 0
```

Comment:
Perhaps instead of using `assert`, due to the fact that it might/will be removed when compiling to optimized byte code i.e. `assert` statements can be globally disabled with the `-O` command-line flag.

Rather consider doing it this way.
```python
if not fake_log.warn.call_count:
    raise AssertionError()
```

Reply:
Just interested in this comment - do you run tests with the optimisation flag? Surely that'll cause almost all tests to be useless? Am I missing something?

Reply:
There are some tests that require a bit of optimization, in those cases, I do run with `-O` flag set and if I need an assertion I just raise it.

Reply:
Thanks. In our case, consistency and having recognizable idioms is far more important than shaving a few seconds off our test suite.

Reply:
@mmphego I understand your point, however I think that this kind of optimization has to be done once and for all in the whole project and a guideline written when this kind of change makes itself necessary.
Changing the check style just for this issue would introduce an heterogeneous style.

WDYT @shin- ?